clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/configuration/CustomConfigClientConfiguration.groovy

```diff
@@ -0,0 +1,107 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.configuration
+
+import org.apache.http.client.HttpClient
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.ssl.SSLContexts
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.cloud.config.client.ConfigClientProperties
+import org.springframework.cloud.config.client.ConfigServicePropertySourceLocator
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.core.env.Environment
+import org.springframework.http.client.ClientHttpRequestFactory
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory
+import org.springframework.web.client.RestTemplate
+
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+import java.security.KeyStore
+import java.security.SecureRandom
+
+/**
+ * Configures Spring Cloud Config Client to speak to a Config Server over an SSL encrypted channel.
+ *
+ * NOTE: By default, spring.cloud.config.enabled=true activates Config Client, but WITHOUT SSL. TO
+ *       have Config Client with SSL, use config-client.ssl.enabled=true instead.
+ *
+ * For property settings, see commented out settings in bootstrap.yml.
+ *
+ * NOTE: This is independent of serving requests via SSL.
+ */
+@Configuration
+@ConditionalOnProperty('config-client.ssl.enabled')
+@EnableConfigurationProperties(ConfigClientSslProperties)
+class CustomConfigClientConfiguration {
+
+	@Autowired
+	ConfigClientSslProperties configClientSslProperties
+
+
+	@Bean
+	ConfigClientProperties configClientProperties(Environment env) {
+		ConfigClientProperties configClientProperties = new ConfigClientProperties(env)
+		configClientProperties.enabled = false
+		configClientProperties
+	}
+
+	@Bean
+	ConfigServicePropertySourceLocator configServicePropertySourceLocator(ConfigClientProperties clientProperties,
+																		  RestTemplate restTemplate) {
+		ConfigServicePropertySourceLocator configServicePropertySourceLocator = new ConfigServicePropertySourceLocator(clientProperties)
+		configServicePropertySourceLocator.restTemplate = restTemplate
+		configServicePropertySourceLocator
+	}
+
+	@Bean
+	RestTemplate restTemplate(ClientHttpRequestFactory clientHttpRequestFactory) {
+		new RestTemplate(clientHttpRequestFactory)
+	}
+
+	@Bean
+	ClientHttpRequestFactory clientHttpRequestFactory(HttpClient httpClient) {
```

Comment:
looks like that needs us to get upgraded past spring 4.1.x 

so I'd recommend bringing in OkHttpClientConfigurationProperties for now and configuring the HttpClient with it instead of the config-server bean, and when we get upgraded we can switch out the HttpComponentsCHRF to the OkHttp flavor